### PR TITLE
Add support for big endian crc32

### PR DESCRIPTION
**Pull request by amock**
This is copied from [1] with minor changes since no support is
needed for reflected crcs, just big-endian and little-endian.

This is work towards fixing #191 

[1] https://github.com/antonblanchard/crc32-vpmsum/

Signed-off-by: Alan Mock <alan@alanmock.com>

### CODE REVIEW

File: lib/crc32_power.c [hunk 1]
```diff
@@ -502,6 +551,9 @@ __crc32_vpmsum(unsigned int crc, const void* p, unsigned long len) {
     /* shift result into top 64 bits of */
     v0 = (__vector unsigned long long)vec_sld((__vector unsigned char)v0,
                                               (__vector unsigned char)vzero, 4);
-
-    return v0[1];
+#ifdef __BIG_ENDIAN__
+    return __builtin_unpack_vector_int128 ((vector __int128_t)(v0), 0);
+#else
+    return __builtin_unpack_vector_int128 ((vector __int128_t)(v0), 1);
+#endif
```
Comment (tuliom) [inline lib/crc32_power.c:558]: This appears to be wrong is causing ppc64le to fail.

**Review by tuliom — CHANGES_REQUESTED** [review]
@amock , the changes look good to me, but I believe they need a few changes:

1. The second patch is unnecessary. Removing trailing whitespace is a plus.
2. We need your `Signed-off-by` in the commit message showing that you agree with https://developercertificate.org.
3. In the commit message, could you elaborate if this implementation is coming from crc32-vpmsum. I noticed some differences and I'm not sure if this is a copy or not.
4. This patch is causing a regression on ppc64le.  Could you check it, please?

**Comment by amock** [comment]
@tuliom thanks for the review.  I think I have made all the requested changes.

File: lib/crc32_power.c [hunk 2]
```diff
@@ -120,6 +130,7 @@ __crc32_vpmsum(unsigned int crc, const void* p, unsigned long len) {
 
         vconst1 = vec_ld(offset, vcrc_short_const);
         vdata0 = vec_ld(0, (__vector unsigned long long*) p);
+        VEC_PERM(vdata0, vdata0, vconst1, vperm_const);
```
Comment (RajalakshmiSR) [inline lib/crc32_power.c:133]: @amock vperm_const is defined only for big-endian. Do we need it for LE, even though VEC_PERM is an empty macro?

Comment (amock) [reply lib/crc32_power.c:133]: We do not need it since it vanishes with the empty macro on LE.

**Review by tuliom — APPROVED** [review]
@amock I'm apologize for taking so long to review it.
LGTM.
@RajalakshmiSR , any other comments?

**Comment by RajalakshmiSR** [comment]
> LGTM.
> @RajalakshmiSR , any other comments?

I merged it, Thanks.

**Pull request merged**